src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -1575,17 +1575,17 @@ fn resolution_failure(
                         _ => None,
                     };
                     // See if this was a module: `[path]` or `[std::io::nope]`
-                    if let Some(module) = last_found_module {
-                        let module_name = collector.cx.tcx.item_name(module);
+                    if let Some(_module) = last_found_module {
                         let note = format!(
-                            "the module `{}` contains no item named `{}`",
-                            module_name, unresolved
+                            "there is no item named `{}` in scope",
```

Comment:
This should only change if there is no intermediate path: `[x]`. If we have `[std::io::nope]` then this is actively misleading. The way to tell that is `partial_res.is_none()`.

Reply:
I have:

```rust
                        let note = if partial_res.is_none() {
                            format!("there is no item named `{}` in scope", unresolved)
                        } else {
                            format!("not in scope")
                        };
```

Which produces this output:

```
error: unresolved link to `path::to::nonexistent::module`
  --> $DIR/intra-link-errors.rs:7:6
   |
LL | /// [path::to::nonexistent::module]
   |      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ there is no item named `path` in scope
   |
note: the lint level is defined here
  --> $DIR/intra-link-errors.rs:1:9
   |
LL | #![deny(broken_intra_doc_links)]
   |         ^^^^^^^^^^^^^^^^^^^^^^

error: unresolved link to `path::to::nonexistent::macro`
  --> $DIR/intra-link-errors.rs:11:6
   |
LL | /// [path::to::nonexistent::macro!]
   |      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ there is no item named `path` in scope

error: unresolved link to `path::to::nonexistent::type`
  --> $DIR/intra-link-errors.rs:15:6
   |
LL | /// [type@path::to::nonexistent::type]
   |      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ there is no item named `path` in scope

error: unresolved link to `std::io::not::here`
  --> $DIR/intra-link-errors.rs:19:6
   |
LL | /// [std::io::not::here]
   |      ^^^^^^^^^^^^^^^^^^ not in scope

error: unresolved link to `std::io::not::here`
  --> $DIR/intra-link-errors.rs:23:6
   |
LL | /// [type@std::io::not::here]
   |      ^^^^^^^^^^^^^^^^^^^^^^^ not in scope
```

What should the messages be?

Reply:
The ones for `path` look fine. I would keep the others the way they are now: 'no item named `not` in module `io`'.